echopype/tests/utils/test_processinglevels_integration.py

```diff
@@ -0,0 +1,121 @@
+    if "longitude" in ed['Platform'].data_vars and "latitude" in ed['Platform'].data_vars:
+        _presence_test(ed["Top-level"], "Level 1A")
+    elif "longitude" in extras and "latitude" in extras:
+        _absence_test(ed["Top-level"])
+        point_ds = xr.Dataset(
+            {
+                "latitude": (["time"], np.array([float(extras["latitude"])])),
+                "longitude": (["time"], np.array([float(extras["longitude"])])),
+            },
+            coords={
+                "time": (["time"], np.array([ed["Sonar/Beam_group1"]["ping_time"].values.min()]))
+            },
+        )
+        ed.update_platform(point_ds)
+        _presence_test(ed["Top-level"], "Level 1A")
+    else:
+        _absence_test(ed["Top-level"])
+        raise RuntimeError(
+            "Platform latitude and longitude are not present and cannot be added "
+            "using update_platform based on test raw file and included parameters."
+        )
+
+    # ---- Calibrate and add_latlon
+    env_params = None
+    if sonar_model == "AZFP":
+        # AZFP data require external salinity and pressure
+        env_params = {
+            "temperature": ed["Environment"]["temperature"].values.mean(),
+            "salinity": extras["salinity"],
+            "pressure": extras["pressure"],
+        }
+
+    ds = ep.calibrate.compute_Sv(echodata=ed, env_params=env_params)
+    _absence_test(ds)
+
+    sv_ds = ep.consolidate.add_location(ds=ds, echodata=ed)
+    assert "longitude" in sv_ds.data_vars and "latitude" in sv_ds.data_vars
+    _presence_test(sv_ds, "Level 2A")
+
+    # ---- Noise removal
+    denoised_ds = ep.clean.remove_noise(sv_ds, ping_num=10, range_sample_num=20)
+    _presence_test(denoised_ds, "Level 2B")
+
+    # ---- apply_mask based on frequency differencing
+    def _freqdiff_applymask(test_ds):
+        # frequency_differencing expects a dataarray variable named "Sv". For denoised Sv,
+        # rename Sv to Sv_raw and Sv_corrected to Sv before passing ds to frequency_differencing
+        if "Sv_corrected" in test_ds.data_vars:
+            out_ds = test_ds.rename_vars(name_dict={"Sv": "Sv_raw", "Sv_corrected": "Sv"})
+        else:
+            out_ds = test_ds
+        freqAB = list(out_ds.frequency_nominal.values[:2])
+        freqdiff_da = ep.mask.frequency_differencing(source_Sv=out_ds, freqAB=freqAB, operator=">", diff=5)
+
+        # Create a new, single-channel Sv variable in ds to pass to apply_mask.
+        # The resulting masked Sv will be single-channel.
+        out_ds["Sv_ch0"] = out_ds["Sv"].isel(channel=0).squeeze()
+        return ep.mask.apply_mask(source_ds=out_ds, var_name="Sv_ch0", mask=freqdiff_da)
+
+    # On Sv w/o noise removal
+    ds = _freqdiff_applymask(sv_ds)
+    _presence_test(ds, "Level 3A")
+
+    # On denoised Sv
+    ds = _freqdiff_applymask(denoised_ds)
+    _presence_test(ds, "Level 3B")
+
+    # ---- Compute MVBS
+    # compute_MVBS expects a variable named "Sv"
```

Comment:
```suggestion
    # compute_MVBS expects a variable named "Sv"
    # No product level is assigned because at present compute_MVBS drops the lat/lon data associated with the input Sv dataset
```
Just adding a clarification, since I was confused for a sec.